### ISSUE TITLE
panic when evaluating a specific regex

### ISSUE THREAD

**Issue by vthib** [fancy-regex/fancy-regex#103]
Using the regex & text used in benches, but with the public `Regex::new` API instead of the private one, leads to a panic in the eval VM:

```rust
use fancy_regex::Regex;

fn main() {
    let regex = Regex::new("^.*?(([ab]+)\\1b)").unwrap();
    eprintln!("match: {:?}", regex.is_match("babab"));
}
```

gives this:

```
thread 'main' panicked at 'byte index 18446744073709551615 is out of bounds of `babab`', library/core/src/str/mod.rs:107:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
   1: core::panicking::panic_fmt
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
   2: core::str::slice_error_fail_rt
   3: core::ops::function::FnOnce::call_once
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:248:5
   4: core::intrinsics::const_eval_select
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/intrinsics.rs:2696:5
   5: core::str::slice_error_fail
             at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/str/mod.rs:86:9
   6: fancy_regex::vm::run
   7: fancy_regex::Regex::is_match
   8: bug::main
```

This is both present on `0.10.0` and on current main (f594a9ba1f675fce87b1f24a26d95d448afacc75)

**Comment by keith-hall** [comment]
Thanks for reporting. This is also reproducible as `cargo run --example toy run '^.*?(([ab]+)\1b)' babab`. It seems like it is somehow caused by the `.*` being lazy.

**Comment by robinst** [comment]
Thanks for reporting! See https://github.com/fancy-regex/fancy-regex/pull/107 for a fix.

**Comment by robinst** [comment]
Fix released in version 0.11.0: https://github.com/fancy-regex/fancy-regex/blob/main/CHANGELOG.md#0110---2023-01-12
